core/sdk/go_sdk.go

```diff
@@ -621,6 +621,12 @@ func (sdk *goSDK) baseWithCodegen(
 		return ctr, fmt.Errorf("failed to mount introspection json file into go module sdk container codegen: %w", err)
 	}
 
+	if cfg := src.Self().SDK; cfg != nil && cfg.Debug {
+		if err := dag.Select(ctx, ctr, &ctr, dagql.Selector{Field: "terminal"}); err != nil {
+			return ctr, fmt.Errorf("failed to enable go sdk runtime terminal: %w", err)
+		}
+	}
+
```

Comment:
In Python you open the terminal right before returning the terminal, in Typescript you do it before setup the container and in Go you do it right in the middle.

I think it should be consistent, maybe at the end to get the full context?

Reply:
agreed, good catch 🙏